universal-login-react/src/ui/Onboarding/Onboarding.tsx

```diff
@@ -36,37 +39,47 @@ export const Onboarding = (props: OnboardingProps) => {
     props.onConnect && props.onConnect();
   };
 
-  const showWalletCreationModal = (transactionHash?: string) => {
-    const relayerConfig = props.sdk.getRelayerConfig();
-    modalService.showModal('waitingForDeployment', {
-      relayerConfig,
-      transactionHash,
-    });
-  };
-
   const onCreateClick = async (ensName: string) => {
-    let gasParameters = INITIAL_GAS_PARAMETERS;
-    const {waitForBalance, contractAddress, privateKey} = await walletService.createFutureWallet();
-    localStorage.setItem('BACKUP_DEMO', JSON.stringify({
-      ensName,
-      contractAddress,
-      privateKey,
-    }));
-    const topUpProps: TopUpProps = {
-      contractAddress,
-      onGasParametersChanged: (parameters: GasParameters) => {gasParameters = parameters;},
-      sdk: props.sdk,
-      isDeployment: true,
-      hideModal: () => {walletService.disconnect(); modalService.hideModal();},
-    };
-    modalService.showModal('topUpAccount', topUpProps);
-    await waitForBalance();
-    showWalletCreationModal();
-    const wallet = await walletService.deployFutureWallet(ensName, gasParameters.gasPrice.toString(), gasParameters.gasToken, showWalletCreationModal);
+    walletCreationService.onBalancePresent(() => modalService.hideModal());
+    const wallet = await walletCreationService.initiateCreationFlow(ensName);
     modalService.hideModal();
     props.onCreate && props.onCreate(wallet);
   };
 
+  const walletState = useProperty(walletService.stateProperty);
+  const transactionHash = useProperty(walletCreationService.deploymentTransactionHash);
+  const renderModal = () => {
```

Comment:
Extract to a separate file and make it component for example `DeploymentModal`

Reply:
👍

Reply:
👍